ES2022 method used in admin code which targets ES2020

### Shopware Version

6.6

### Affected area / extension

Platform(Default)

### Actual behaviour

The admin panel’s `tsconfig.json` targets ES2020 and imports the standard library from ES2020.

However, the code uses two calls to `Array#at` , which was only added in ES2022:
https://github.com/shopware/shopware/blob/b0500dd6ad0c99d9e11be7a98ccb9238c779c733/src/Administration/Resources/app/administration/src/app/service/file-validation.service.ts#L109
https://github.com/shopware/shopware/blob/b0500dd6ad0c99d9e11be7a98ccb9238c779c733/src/Administration/Resources/app/administration/src/app/service/file-validation.service.ts#L142

tsc does not detect this as an error because a declaration for this method is present inside `@types/node`, which gets imported through multiple admin scripts. I only spotted this error by running `tsc` in my admin plugin’s folder, probably because it does not import enough admin code to inhale those type declarations.

### Expected behaviour

Ideally, the project should either target ES2022+ or avoid using code that does not work in ES2020.

### How to reproduce

Create a `tsconfig.json` that copies the admin panel’s settings and imports the bare minimum part of the admin panel, try to compile it using `tsc`:

```
{
  "compilerOptions": {
    "target": "es2020",
    "lib": ["ES2020", "DOM"],
    "module": "commonjs",
    "resolveJsonModule": true,
    "allowJs": true,
    "noEmit": true,
    "isolatedModules": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true,
    "noErrorTruncation": false,
    "paths": {
      "src/*": ["../../../../../../../vendor/shopware/administration/Resources/app/administration/src/*"]
    },
  },
  "exclude": [
    "node_modules"
  ],
  "include": [
    "../../../../../../../vendor/shopware/administration/Resources/app/administration/src/entity-schema-definition.d.ts",
    "../../../../../../../vendor/shopware/administration/Resources/app/administration/src/global.types.ts",
    "../../../../../../../vendor/shopware/administration/Resources/app/administration/src/html-shim.d.ts",
  ]
}

```

It produces two errors complaining about this method:

```
../../../../../../../vendor/shopware/administration/Resources/app/administration/src/app/service/file-validation.service.ts:109:63 - error TS2550: Property 'at' does not exist on type 'string[]'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2022' or later.

109             const currentFileExtension = file.name.split('.').at(-1);
                                                                  ~~

../../../../../../../vendor/shopware/administration/Resources/app/administration/src/app/service/file-validation.service.ts:142:76 - error TS2550: Property 'at' does not exist on type 'string[]'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2022' or later.

142             if (mimeAccept === 'model/gltf-binary' && file.name.split('.').at(-1) === 'glb' && file.type === '') {
                                                                               ~~

Found 2 errors in the same file, starting at: ../../../../../../../vendor/shopware/administration/Resources/app/administration/src/app/service/file-validation.service.ts:109
```


### Definition of Done

- [ ] An automated test covers the fix.
- [ ] Integration/E2E testing in staging is done.
- [ ] Changelog markdown file created or updated in /changelog/_unreleased directory.
- [ ] Developer documentation written/updated.
- [ ] End user and/or developer documentation written or updated with clarification.
- [ ] Regression test created to ensure bug does not reappear, and automated regression tests pass.
- [ ] Code changes resolve the reported bug without introducing new issues.

--- a/src/Administration/Resources/app/administration/tsconfig.json
+++ b/src/Administration/Resources/app/administration/tsconfig.json
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
-    "target": "es2020",
-    "lib": ["ES2020", "DOM"],
+    "target": "ES2023",
+    "lib": ["ES2020", "DOM", "ES2023"],
     "module": "commonjs",
     "rootDir": "./",
     "baseUrl": "./",

